torch/csrc/jit/script/module.h

```diff
@@ -0,0 +1,246 @@
+#pragma once
+#include "torch/csrc/jit/ir.h"
+#include "torch/csrc/jit/graph_executor.h"
+#include "torch/csrc/autograd/variable.h"
+
+namespace torch { namespace jit { namespace script {
+
+struct Method {
+  Method(std::string name, bool optimize)
+  : name_(std::move(name))
+  , graph_(std::make_shared<Graph>())
+  , optimize(optimize) {}
+  variable_tensor_list run(variable_tensor_list && inputs) {
+    std::call_once(executor_init, [&]{
+      executor = GraphExecutor(graph_, optimize);
+    });
+    for(auto tp : member_inputs) {
+      inputs.push_back(*tp);
+    }
+    return executor.run(std::move(inputs));
+  }
+  std::shared_ptr<Graph> graph() const {
+    return graph_;
+  }
+
+  const std::string & name() const {
+    return name_;
+  }
+  // emit a function call by inlining the callees Graph into this one
+  // adding any extra parameters necessary to do this call
+
+  // defined here to keep details of member_input handling confined to this class
+  std::vector<Value*> emit_call_to(Method & callee, ArrayRef<Value*> inputs) {
+    JIT_ASSERT(!executor);
+    auto fn = callee.graph();
```

Comment:
It's annoying that this is not generic inlining, because of the `callee.member_inputs` business. Arguably "member inputs" are just free variables, so maybe you can get some generic here.